tests/RoyalmailTest.php

```diff
@@ -185,4 +424,67 @@ public function testRoyalMailClassNull()
         $this->assertEquals($this->emptyArray, $this->calculateMethodClass->getMethods(null, null, null));
     }
 
+    /**
+     * Test co compare the meta names vs the clean name to check that the correct value
+     * exists and is being used.
+     */
+    public function testMethodToMetaVsCleanName()
+    {
+        foreach ($this->dataClass->mappingMethodToMeta as $array => $data){
+            foreach ($this->dataClass->mappingCleanNameToMethod as $method => $methodData)
+            {
+                if ($data[0] ==  $methodData[2]){
+                    $this->assertEquals($data[4], $methodData[1]);
+                }
+            }
+        }
+    }
+
+    /**
+     * Test for insurance value checking that the correct insurance value is being
+     * used in the CSV files
+     */
+    public function testInsuranceValue()
+    {
+        foreach ($this->dataClass->mappingMethodToMeta as $array => $data){
+            foreach ($this->dataClass->mappingDeliveryToPrice as $method => $methodData)
+            {
+                if ($data[0] == $methodData[0]){
+                    if ( $methodData[4] != ""){
+                        $this->assertEquals($data[3], $methodData[4]);
+                        print_r($data, $method);
+                    }
+                }
+            }
+        }
+
+        foreach ($this->dataClass->mappingMethodToMeta as $array => $data){
+            foreach ($this->dataClass->mappingCleanNameToMethod as $method => $methodData) {
+                if ($data[0] == $methodData[2]) {
+                    if ($methodData[5] != ""){
+                        $this->assertEquals($data[3], $methodData[5]);
+                        print_r($data, $method);
+                    }
+                }
+            }
+
+        }
+    }
+
+    /**
+     * Test to compare the method clean name vs the method group, ensuring that
+     * the clean names are correct and exists.
+     */
+    public function testCleanNameVsMethodGroup()
+    {
+        foreach ($this->dataClass->mappingCleanNameToMethod as $array => $data){
```

Comment:
$array isn't used so this can just be  "as $data". Even if it was it would be a key not an array which would be confusing.